src/wrapper/SquidexClient.ts

```diff
@@ -0,0 +1,69 @@
+import { SquidexClient as FernClient } from "../Client";
+import * as environments from "../environments";
+import * as core from "../core";
+import * as errors from "../errors";
+import urlJoin from "url-join";
+
+export declare namespace SquidexClient {
+    interface Options {
+        clientId: string;
+        clientSecret: string;
+        environment?: environments.SquidexEnvironment | string;
+    }
+}
+
+export class SquidexClient extends FernClient {
+    private token: string | undefined;
+
+    constructor(options: SquidexClient.Options) {
+        super({
+            environment: options.environment,
+            token: async () => {
+                if (this.token == null) {
+                    const response = await core.fetcher({
+                        url: urlJoin(
+                            this.options.environment ?? environments.SquidexEnvironment.Default,
+                            "/identity-server/connect/token"
+                        ),
+                        contentType: "application/x-www-form-urlencoded",
+                        method: "POST",
+                        body: {
+                            grant_type: "client_credentials",
+                            client_id: options.clientId,
+                            client_secret: options.clientSecret,
+                            scope: "squidex-api",
+                        },
+                    });
+                    if (response.ok) {
+                        if (typeof response.body !== "string") {
+                            throw new errors.SquidexError({
+                                message: "Token is not a string",
+                            });
+                        }
+                        this.token = response.body;
```

Comment:
I think the token is a not a string but a JSON object.

Reply:
See: https://connect2id.com/products/server/docs/api/token#token-response

Reply:
makes sense so really we want `response.body.accessToken` ?

Reply:
`access_token` ... but yes.

Reply:
fix is here https://github.com/Squidex/sdk-node/pull/4